### DISCUSSION TITLE
P4VFS and WSL2 9p filesystem

### DISCUSSION THREAD

**Discussion by lweiner-spirent** [microsoft/p4vfs#30] [Q&A]
I have used P4VFS on Windows to sync to the latest version of my project (`p4vfs sync ...#head`).  However, I'm using VSCode running under WSL2/Ubuntu-20.04 to edit files.  When I try to access the files using VSCode or Ubuntu command line, then I only see zeros in the file:

If I use a native Windows program to read the file, then I can see the actual contents of the file in WSL...

```
$ od -tx1 SourceCode.cc
0000000 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
*
0213720 00 00 00 00 00 00 00 00 00 00 00
0213733

$ /mnt/c/windows/system32/cmd.exe <<< "type sourceCode.cc" &> /dev/null
$ od -tx1 SourceCode.cc
0000000  2f  2f  20  24  49  64  3a  20  ...
...
0213720  53  74  61  74  65  3b  0a  7d  0a 
0213733
```

Is it possible to make this run seamlessly under WSL?  (i.e. without the need to use a windows native program to trigger P4VFS)?

**Comment by jessk-msft** [comment]
Thanks for calling this out. It looks like the filter driver is being skipped for non-windows-native processes. In this case you're seeing zeros which have padded out the SPARSE stream of the P4VFS placeholder file. I actually haven't considered WSL until now, but it would be great to properly support it. 

I'll dig around to see how our most common filters behave with WSL.
Regards
-Jess
